src/command_modules/azure-cli-acr/azure/cli/command_modules/acr/pack.py

```diff
@@ -0,0 +1,100 @@
+# --------------------------------------------------------------------------------------------
+# Copyright (c) Microsoft Corporation. All rights reserved.
+# Licensed under the MIT License. See License.txt in the project root for license information.
+# --------------------------------------------------------------------------------------------
+
+import os
+import uuid
+import tempfile
+from knack.log import get_logger
+from knack.util import CLIError
+from azure.cli.core.commands import LongRunningOperation
+
+from ._constants import ACR_TASK_YAML_DEFAULT_NAME, ORYX_PACK_BUILDER_IMAGE
+from ._stream_utils import stream_logs
+from ._utils import (
+    validate_managed_registry,
+    get_validate_platform,
+    get_custom_registry_credentials,
+    get_yaml_and_values
+)
+from ._client_factory import cf_acr_registries
+from ._archive_utils import upload_source_code, check_remote_source_code
+from .run import prepare_source_location
```

Comment:
>prepare_source_location [](start = 17, length = 23)

I'm not sure if you can just reuse the same code as run if pack doesn't support /dev/null.

Reply:
Why not? Won't it just mean the first condition in `prepare_source_location` (`if source_location.lower() == NULL_SOURCE_LOCATION`) will always be skipped?

Reply:
If pack command doesn't support `az acr pack /dev/null`, it's better to block it on client. Otherwise the request will be sent to service and we will allocate resource to execute it. User will be charged for the resource.
If you want to reuse the function, it's better to validate `/dev/null` and return error before you call the function.

Reply:
this is what happens now though - if `prepare_source_location` returns None, like it would if the source location is `/dev/null`, a CLIError will be raised (`pack.py:53`)

Reply:
Great. Thanks

---
In reply to: [285774935](https://github.com/Azure/azure-cli/pull/9366#discussion_r285774935) [](ancestors = 285774935)